EssentialCSharp.Web/wwwroot/js/site.js

```diff
@@ -78,194 +78,226 @@ const completedFeaturesList = [
+                }
+            }
+        });
+
+        const sidebarShown = ref(false);
+
+        const smallScreen = computed(() => {
+            return (windowWidth.value || 0) < smallScreenSize;
+        });
+
+        /** @type {import("vue").Ref<"toc" | "search">} */
+        const sidebarTab = ref("toc");
+
+        const currentPage = findCurrentPage([], tocData) ?? [];
+
+        const chapterParentPage = currentPage.find((parent) => parent.level === 0);
+
+        const sectionTitle = ref(currentPage?.[0]?.title || "Essential C#");
+        const expandedTocs = reactive(new Set());
+        for (const item of currentPage) {
+            expandedTocs.add(item.key);
+        }
+
+        // hide the sidebar when resizing to small screen
+        // to do: make it re emerge when going from small to big
+        watch(windowWidth, (newWidth, oldWidth) => {
+            //+ 50 so that the side bar diappears before the css media class changes the sidebar to take
+            // over the full screen
+            if (newWidth < smallScreenSize) {
+                sidebarShown.value = false;
+            }
+            // when making screen bigger reveal sidebar
+            else {
+                if (!sidebarShown.value) {
+                    sidebarShown.value = true;
+                }
+            }
+        });
+
+        // prevent scrolling of the page when the sidebar is visible on small screens
+        watch(sidebarShown, (newValue, oldValue) => {
+            if (windowWidth.value <= smallScreenSize) {
+                if (newValue) {
+                    document.body.classList.add("noScrollOnSmallScreen");
+                } else {
+                    document.body.classList.remove("noScrollOnSmallScreen");
+                }
+            }
+        });
+
+        onMounted(() => {
+            if (windowWidth.value > smallScreenSize) {
+                sidebarShown.value = true;
+            }
+
+            // Scroll the current selected page in the TOC into view of the TOC.
+            [...document.querySelectorAll(".current-section")]
+                .reverse()[0]
+                ?.scrollIntoView({
+                    behavior: "auto",
+                    block: "center",
+                    inline: "center",
+                });
+        });
 
-    // hide the sidebar when resizing to small screen
-    // to do: make it re emerge when going from small to big
-    watch(windowWidth, (newWidth, oldWidth) => {
-      //+ 50 so that the side bar diappears before the css media class changes the sidebar to take
-      // over the full screen
-      if (newWidth < smallScreenSize) {
-        sidebarShown.value = false;
-      }
-      // when making screen bigger reveal sidebar
-      else {
-        if (!sidebarShown.value) {
-          sidebarShown.value = true;
+        const enableTocFilter = ref('none');
+        const searchQuery = ref('');
+
+        const filteredTocData = computed(() => {

```

Comment:
Performing stateful mutations in a computed is unusual and leads to confusing spaghetti. Use a `watchEffect` instead, or remove mutations of `expandedTocs`.

Reply:
I think I have now resolved that if this looks correct? https://github.com/IntelliTect/EssentialCSharp.Web/pull/585/commits/496159828c4f957bdc4fd8b67e17384d559015a2

Reply:
There's still a stateful mutation in a computed on line 243.

I probably should have actually advised that you use a simple `watch` against `searchQuery`.

Reply:
Thanks for catching my obvious oversight :) Here is another crack at it: https://github.com/IntelliTect/EssentialCSharp.Web/pull/585/commits/f475110b8c637e14101f73c3aa207f32f0053013